### PR TITLE
update store on updating attributes in cluster.go

### PR DESCRIPTION
**Pull request by il9ue**
Added logic that updates store with given attributes.

### CODE REVIEW

**Comment by gyuho** [comment]
@il9ue Test fails and please `gofmt` your code. And please follow our contribution guide https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message.

Thanks!

**Comment by il9ue** [comment]
@gyuho Changed few lines in proper logic and made sure passed test and gofmt! Sorry for few mistakes

Thanks!

File: etcdserver/cluster.go [hunk 1]
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
...
 	if ok {
 		plog.Warningf("skipped updating attributes of removed member %s", id)
 	} else {
+		if _, err := c.store.Update(p, string(b), store.Permanent); err != nil {
+			plog.Panicf("update Attributes should never fail: %v", err)
+		} else {
+			c.members[id].Attributes = attr
+			return true
+		}
 		plog.Panicf("error updating attributes of unknown member %s", id)
```
Comment (gyuho) [inline etcdserver/cluster.go]: This line is not needed anymore, right?

**Comment by gyuho** [comment]
This looks good, but defer to @heyitsanthony and @xiang90 for final reviews.

Thanks,

**Comment by gyuho** [comment]
@il9ue  As explained in https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message, please change the PR and commit message to `etcdserver: update store on updating attributes` and squash two commits into one because they are doing one change for one file.

File: etcdserver/cluster.go [hunk 2]
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
...
 		return true
 	}
+	b, err := json.Marshal(attr)
+	if err != nil {
+		plog.Panicf("marshal attributes should never fail: %v", err)
+	}
+	p := path.Join(memberStoreKey(id), attributesSuffix)
 	_, ok := c.removed[id]
 	if ok {
 		plog.Warningf("skipped updating attributes of removed member %s", id)
```
Comment (heyitsanthony) [inline etcdserver/cluster.go]: the if/else nesting can be helped a bit by `return false` after the `Warningf()`, then removing the following `else`

File: etcdserver/cluster.go [hunk 3]
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
...
 	} else {
+		if _, err := c.store.Update(p, string(b), store.Permanent); err != nil {
+			plog.Panicf("update Attributes should never fail: %v", err)
+		} else {
```
Comment (heyitsanthony) [inline etcdserver/cluster.go]: `else` is unnecessary here too because of the panic

**Comment by il9ue** [comment]
@gyuho updated the commit message and squashed two previous commits. Also removed lines after 
```
Panicf()
```
@heyitsanthony injected 
```
return false
```
just to clarify logic, thanks!

File: etcdserver/cluster.go [hunk 4]
```diff
@@ -334,13 +334,20 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
...
 		plog.Warningf("skipped updating attributes of removed member %s", id)
+		return false
 	} else {
-		plog.Panicf("error updating attributes of unknown member %s", id)
+		if _, err := c.store.Update(p, string(b), store.Permanent); err != nil {
+			plog.Panicf("update Attributes should never fail: %v", err)
+		}
 	}
-	// TODO: update store in this function
 	return false
```
Comment (heyitsanthony) [inline etcdserver/cluster.go]: `return true` at the end since the update succeeded

**Comment by heyitsanthony** [comment]
fixup the success return value and it's lgtm. thanks!

**Comment by il9ue** [comment]
@heyitsanthony  injected ```return true``` for success. Thanks!

**Comment by xiang90** [comment]
@il9ue We actually update the attribute here https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L1044 now. We need to avoid this if you move the updating store to cluster.go

**Comment by il9ue** [comment]
Will check why build failed, could you give me feedback to resolve errors? It passed when tested locally, Thanks!

**Pull request closed without merging**